diff --git a/src/main/java/com/outscale/osc_sdk_java/client/ConfigurationEnv.java b/src/main/java/com/outscale/osc_sdk_java/client/ConfigurationEnv.java
new file mode 100644
index 0000000..c1021d3
--- /dev/null
+++ b/src/main/java/com/outscale/osc_sdk_java/client/ConfigurationEnv.java
@@ -0,0 +1,39 @@
+package com.outscale.osc_sdk_java.client;
+
+public class ConfigurationEnv extends ConfigurationInitializer {
+    private Profile profile;
+
+    public ConfigurationEnv() {
+    }
+
+    public static ConfigurationEnv loadConfigEnv() {
+        ConfigurationEnv configurationEnv = new ConfigurationEnv();
+
+        configurationEnv.profile = new Profile();
+        configurationEnv.profile.setAccessKey(System.getenv("OSC_ACCESS_KEY"));
+        configurationEnv.profile.setSecretKey(System.getenv("OSC_SECRET_KEY"));
+        configurationEnv.profile.setRegion(System.getenv("OSC_REGION"));
+        configurationEnv.profile.setX509ClientCert(System.getenv("OSC_X509_CLIENT_CERT"));
+        configurationEnv.profile.setX509ClientCertB64(System.getenv("OSC_X509_CLIENT_CERT_B64"));
+        configurationEnv.profile.setX509ClientKey(System.getenv("OSC_X509_CLIENT_KEY"));
+        configurationEnv.profile.setX509ClientKeyB64(System.getenv("OSC_X509_CLIENT_KEY_B64"));
+        configurationEnv.profile.setMethod(System.getenv("OSC_METHOD"));
+        configurationEnv.profile.setProtocol(System.getenv("OSC_PROTOCOL"));
+
+        Endpoint endpoints = new Endpoint();
+        endpoints.setApi(System.getenv("OSC_ENDPOINT_API"));
+        configurationEnv.profile.setEndpoints(endpoints);
+
+        return configurationEnv;
+    }
+
+    public ApiClient getApiClient() {
+        return getApiClient(profile);
+    }
+
+    @Override
+    public String toString() {
+        return profile.toString();
+    }
+
+}
diff --git a/src/main/java/com/outscale/osc_sdk_java/client/Endpoint.java b/src/main/java/com/outscale/osc_sdk_java/client/Endpoint.java
index b635fcf..c5d6aca 100644
--- a/src/main/java/com/outscale/osc_sdk_java/client/Endpoint.java
+++ b/src/main/java/com/outscale/osc_sdk_java/client/Endpoint.java
@@ -69,4 +69,8 @@ public class Endpoint {
     public String getApi() {
         return api;
     }
+
+    public void setApi(String api) {
+        this.api = api;
+    }
 }
diff --git a/src/main/java/com/outscale/osc_sdk_java/client/Profile.java b/src/main/java/com/outscale/osc_sdk_java/client/Profile.java
index 9dc7fa7..bbc2f42 100644
--- a/src/main/java/com/outscale/osc_sdk_java/client/Profile.java
+++ b/src/main/java/com/outscale/osc_sdk_java/client/Profile.java
@@ -155,4 +155,44 @@ public class Profile {
     public String getX509ClientKeyB64() {
         return x509ClientKeyB64;
     }
+
+    public void setAccessKey(String accessKey) {
+        this.accessKey = accessKey;
+    }
+
+    public void setEndpoints(Endpoint endpoints) {
+        this.endpoints = endpoints;
+    }
+
+    public void setMethod(String method) {
+        this.method = method;
+    }
+
+    public void setProtocol(String protocol) {
+        this.protocol = protocol;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    public void setSecretKey(String secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    public void setX509ClientCert(String x509ClientCert) {
+        this.x509ClientCert = x509ClientCert;
+    }
+
+    public void setX509ClientCertB64(String x509ClientCertB64) {
+        this.x509ClientCertB64 = x509ClientCertB64;
+    }
+
+    public void setX509ClientKey(String x509ClientKey) {
+        this.x509ClientKey = x509ClientKey;
+    }
+
+    public void setX509ClientKeyB64(String x509ClientKeyB64) {
+        this.x509ClientKeyB64 = x509ClientKeyB64;
+    }
 }
